src/Board.cpp

```diff
@@ -14,4 +18,41 @@ Board::Board(int rowNumber,
     board_.assign(rowNumber_, singleRow);
 }
 
+int getRowNumberFromFile(std::string path)
+{
+    std::ifstream inputFile(path);
+    json j;
+    inputFile >> j;
+    return j["rowNumber"];
+}
+
+int getColNumberFromFile(std::string path)
+{
+    std::ifstream inputFile(path);
+    json j;
+    inputFile >> j;
+    return j["colNumber"];
+}
+
+Board::Board(std::string path)
+    : rowNumber_(getRowNumberFromFile(path)),
+      colNumber_(getColNumberFromFile(path))
+{
+    std::ifstream inputFile(path);
+    json j;
+    inputFile >> j;
+
+    if ((rowNumber_ < 3) or (colNumber_ < 3))
+        throw InvalidDimensions();
+
+    Lines rows = j["rows"];
+    Lines cols = j["cols"];
+    rows_ = rows;
```

Comment:
Why not immediately assign it like:
```
rows_ = j["rows"];
```

Reply:
error: ambiguous overload for ‘operator=’ (operand types are ‘Lines {aka std::vector<std::vector<int> >}’ and ‘nlohmann::basic_json<>::value_type {aka nlohmann::basic_json<>}’)
     rows_ = j["rows"];

Says it all :D rows_(j["rows"]); also fails :/